Use latest version of rack-test

epergo opened the pull request:
Allow versions greater (and equals) than 2.0 for the `rack-test` gem. The only change needed was to remove the `build_rack_test_session` in test_helper because it uses the deprecated `rack_mock_session` which calls `build_rack_test_session` again creating an infinite loop

epergo commented on sinatra.gemspec:
```diff
@@ -46,4 +46,6 @@ EOF
   s.add_dependency 'rack-protection', version
   s.add_dependency 'mustermann', '~> 3.0'
+
+  s.add_development_dependency 'rack-test', '~> 2'
```
I've moved the dependency here so all gemspecs define it in the same place

jkowens commented:
@epergo did you see @jeremyevans' [comment](https://github.com/sinatra/sinatra/discussions/1782#discussioncomment-3157648) about the `default_env` method?

epergo commented:
> @epergo did you see @jeremyevans' [comment](https://github.com/sinatra/sinatra/discussions/1782#discussioncomment-3157648) about the `default_env` method?

:sweat_smile: Nope, didn't see it. The second issue would need to be addressed. I'm having a look and in all tests `global_env` is empty, do you know if it is used somehow? It looks like that it isn't and the custom `Session` class could be dropped altogether.

epergo commented:
I've removed the `Session` custom class so we use the one provided by `rack-test` because we weren't using the extra functionality. Also I've added the last version of `rack-test` from the repository to check if there is any other failure. All steps are working

jkowens commented:
Looks good, thanks!

rondales commented:
> Allow versions greater (and equals) than 2.0 for the `rack-test` gem. The only change needed was to remove the `build_rack_test_`
